samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Activities/ContentActivities.cs

```diff
@@ -0,0 +1,335 @@
+            logger.LogError(ex, "Error researching topic: {Topic}", topic);
+            throw;
+        }
+    }
+}
+
+/// <summary>
+/// Activity to create article content using the content generation agent with knowledge files
+/// </summary>
+[DurableTask]
+public class CreateArticleActivity(ContentGenerationAgentService contentGenerationService, ILogger<CreateArticleActivity> logger)
+    : TaskActivity<(string Topic, ResearchData ResearchData), string>
+{
+
+    public override async Task<string> RunAsync(TaskActivityContext context, (string Topic, ResearchData ResearchData) input)
+    {
+        logger.LogInformation("Creating article for topic: {Topic}", input.Topic);
+
+        try
+        {
+            // Serialize research data back to JSON to pass to agent
+            string researchJson = System.Text.Json.JsonSerializer.Serialize(input.ResearchData, 
+                new System.Text.Json.JsonSerializerOptions { WriteIndented = true });
+            
+            string articleContent = await contentGenerationService.CreateArticleAsync(input.Topic, researchJson);
+            logger.LogInformation("Successfully created article content of {Length} characters", articleContent.Length);
+            return articleContent;
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Error creating article for topic: {Topic}", input.Topic);
+            throw;
+        }
+    }
+}
+
+/// <summary>
+/// Activity to generate images using the image generation agent with DALL-E
+/// </summary>
+[DurableTask]
+public class GenerateImagesActivity(ImageGenerationAgentService imageGenerationService, ILogger<GenerateImagesActivity> logger)
+    : TaskActivity<(string Topic, string ArticleContent), List<GeneratedImage>>
+{
+
+    public override async Task<List<GeneratedImage>> RunAsync(TaskActivityContext context, (string Topic, string ArticleContent) input)
+    {
+        logger.LogInformation("Generating images for article on topic: {Topic}", input.Topic);
+
+        try
+        {
+            string imagesJson = await imageGenerationService.GenerateImagesAsync(input.Topic, input.ArticleContent);
+            logger.LogInformation("Successfully generated image data");
+            
+            // Clean the JSON response
+            string cleanJson = imageGenerationService.CleanJsonResponse(imagesJson);
+            
+            // Parse the JSON into generated images
+            List<GeneratedImage> images = GeneratedImage.FromJson(cleanJson);
+            return images;
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "Error generating images for topic: {Topic}", input.Topic);
+            throw;
+        }
+    }
+}
+
+/// <summary>
+/// Activity to assemble the final article with images in HTML format and save to file
+/// </summary>
+[DurableTask]
+public class AssembleFinalArticleActivity(ILogger<AssembleFinalArticleActivity> logger, string? outputDirectory = null)
```

Comment:
`outputDirectory` will always be `null` (as there's not a way to automatically inject the string)?  Alternatively, it inject `IConfiguration` and pull the output from a configuration value.